services/processor/tasks/genesis/genesis.go

```diff
@@ -1,23 +1,21 @@
 package genesis
 
 import (
-	"bytes"
 	"context"
 	"strconv"
 
 	"github.com/gocraft/work"
 	"github.com/gomodule/redigo/redis"
 	"github.com/ipfs/go-cid"
 	logging "github.com/ipfs/go-log/v2"
-	typegen "github.com/whyrusleeping/cbor-gen"
 	"go.opentelemetry.io/otel/api/global"
 
 	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	init_ "github.com/filecoin-project/lotus/chain/actors/builtin/init"
+	"github.com/filecoin-project/lotus/chain/actors/builtin/miner"
 	"github.com/filecoin-project/lotus/chain/types"
 	"github.com/filecoin-project/specs-actors/actors/builtin"
-	init_ "github.com/filecoin-project/specs-actors/actors/builtin/init"
-	"github.com/filecoin-project/specs-actors/actors/builtin/miner"
-	"github.com/filecoin-project/specs-actors/actors/util/adt"
```

Comment:
The main change here is to the use interface presented by lotus that handles the specs-actors upgrade case.